Seneca-CDOT/starchart
# Add unit tests for Nodemailer

## Description
Resolves #126 by adding initial Unit Tests to test the functionality of Nodemailer. I had to adjust the `sendNotification` function for unit testing. Also, I added `dotenv` config statement in `notifications.server.ts` to read environment variables defined in `.env.example`. 

### Steps to Test
1. Setup the project locally (make sure Docker is running)
2. Run `npm test` *(all tests should pass)*
3. Navigate to the MailHog interface at http://localhost:8025 to see the test notification:
<img width="1447" alt="List view" src="https://user-images.githubusercontent.com/50856799/218288776-5920f66d-fac7-45e3-aaf2-fbbf4940603d.png">
<i>Clicking on the email should navigate to the detailed view:</i>
<img width="1447" alt="Detailed view" src="https://user-images.githubusercontent.com/50856799/218288772-4eb1a710-4fca-4c78-841c-4f02cce0c91f.png">
